### PR TITLE
[android] Add kotlin spotless ktlint linter

### PR DESCRIPTION
**Pull request by wschurman**
# Why

This PR adds [spotless](https://github.com/diffplug/spotless/) with [ktlint](https://github.com/pinterest/ktlint) kotlin linter and formatter. The goal is to establish one set of codestyles from the official style guide with a few exceptions for patterns we're already doing:
- 2 spaces indent, 4 space continuation indent
- allow wildcard imports
- no import sorting (we have stuff around ABI imports that is incompatible with this)

This should hopefully result in less uncertainty about the correct style and less attention cost for PR reviewers (no longer have to review style).

Note that the last thing missing is a good way for us to share sensible IDE defaults (this is missing in www as well though which also has a linter). I tend to do the following:
```
brew install ktlint
ktlint --android applyToIDEAProject
```

# How

This is broken down into 3 commits:
1. Add spotless plugin, configure ktlint in accordance with what our repo already seems to do.
2. Add CI step
3. Fix current lint errors

# Test Plan

Wait for CI which runs the check.

### CODE REVIEW

**Review by ide — APPROVED** [review]
Is there a way to programmatically get the output of spotless? I believe GHA allows for inline warnings & errors (https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message) if stdout contains sentinel tokens.

File: android/build.gradle [hunk 1]
```diff
@@ -102,6 +107,19 @@ allprojects {
...
-  delete rootProject.buildDir
+subprojects {
+  plugins.apply("com.diffplug.spotless")
+  spotless {
+    kotlin {
+      target '**/*.kt'
+      ktlint().userData([
+              "disabled_rules": "no-wildcard-imports,import-ordering",
+              "indent_size": "2",
+              "continuation_indent_size": "4"
```
[the remark this answers is not in the document]
Comment (wschurman) [reply android/build.gradle:118]: We haven't added a groovy style section yet but spotless has that configuration in addition to java as well.

**Comment by wschurman** [comment]
> Is there a way to programmatically get the output of spotless? I believe GHA allows for inline warnings & errors (https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message) if stdout contains sentinel tokens.

I don't see anything in the configuration for a parseable output in spotless, but we can investigate further or maybe write our own upstream. ktlint itself does support different output formats.

For reference, the output for now looks like the following:
```
Execution failed for task ':expo-permissions:spotlessKotlinCheck'.
> The following files had format violations:
      ../packages/expo-permissions/android/src/main/java/expo/modules/permissions/requesters/BackgroundLocationRequester.kt
          @@ -7,7 +7,7 @@
           import·org.unimodules.interfaces.permissions.PermissionsResponse
           import·org.unimodules.interfaces.permissions.PermissionsStatus

          -class·BackgroundLocationRequester:·PermissionRequester·{
          +class·BackgroundLocationRequester·:·PermissionRequester·{
           ··override·fun·parseAndroidPermissions(permissionsResponse:·Map<String,·PermissionsResponse>)·=·when·{
           ····Build.VERSION.SDK_INT·>=·Build.VERSION_CODES.R·->·parseAndroidPermissionsForAndroidR(permissionsResponse)
           ····Build.VERSION.SDK_INT·==·Build.VERSION_CODES.Q·->·parseAndroidPermissionsForAndroidQ(permissionsResponse)
  Run './gradlew :expo-permissions:spotlessApply' to fix these violations.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 16s
172 actionable tasks: 2 executed, 170 up-to-date
::error Spotless lint failed. Run `./gradlew spotlessApply` to automatically fix formatting.
```

**Review by ide — COMMENTED** [review]
The inline comments help since they don’t block tests and are easier to read without having to click through to the CI results. For now, though, what do you think about running the linter after tests? That way serious issues are highlighted first and an extra new line doesn’t block someone from getting test results.

**Review by ide — APPROVED** [review]

**Pull request merged**